src/pudl/io_managers.py

```diff
@@ -87,6 +87,7 @@ def __init__(
         base_dir: str,
         db_name: str,
         md: sa.MetaData = None,
+        timeout: float = 1_000.0,
```

Comment:
Wow, great that it's this easy!

Reply:
What does it mean that the timeout is 1000 seconds? If it doesn't make a connection immediately does it then wait all 1000 seconds? Or does it keep trying periodically for up to 1000 seconds?

Reply:
My guess is we'll eventually end up with some combination of duckdb and Parquet for performance and easy distribution. Once the Arrow backed pandas dataframes are mature we should give that + Parquet a shot, since there shouldn't be any translation required between data types, the types will much richer, and ultimately we'll just be distributing files in cloud buckets that folks can query with DuckDB or Dask or Pandas or whatever.

Maybe we eventually end up with a big collection of files locally that are produced by Dagster, but we only distribute the denormalized tables, so there aren't a large number of files?

Anyway, lots of stuff to think about with someone who knows more than we do and is interested in open data warehousing models!